src/Model/Scope/Condition.php

```diff
@@ -128,6 +128,24 @@ public function __construct($key, $operator = null, $value = null)
                     ->addMoreInfo('operator', $operator);
             }
         }
+
+        if (is_array($value)) {
+            if (array_filter($value, 'is_array')) {
+                throw (new Exception('Multi-dimensional array as condition value is not supported'))
+                    ->addMoreInfo('value', $value);
+            }
+
+            if (!in_array($this->operator, [
+                self::OPERATOR_EQUALS,
```

Comment:
what is the difference between EQUALS and IN operators now?

Reply:
Equivalent in case of array.

Reply:
hmm, are we used to use array with `=`?

Am I correct, that `null = null` will match now, but `null = [null]` will be now converted later to `null in (null)`, thus not work as expected?

if so. we should not alow array with `=` and definitelly `in` should accept only single level array

Reply:
> Am I correct, that `null = null` will match now, but `null = [null]` will be now converted later to null in (null), thus not work as expected?

This is quite edge case. Result of `null = null` is not what you are implying I believe
https://stackoverflow.com/questions/1843451/why-does-null-null-evaluate-to-false-in-sql-server

> `in` should accept only single level array

Done

But we will keep equivalence of `=` and `in`. `in` also works with non-array value same as `=`

Merging

Reply:
> > Am I correct, that `null = null` will match now, but `null = [null]` will be now converted later to null in (null), thus not work as expected?
> 
> This is quite edge case. Result of `null = null` is not what you are implying I believe
> https://stackoverflow.com/questions/1843451/why-does-null-null-evaluate-to-false-in-sql-server

no, no, how WE support it, as far as I know we convert `= null` to `is null`, so `= [null]` MUST NEVER BEHAVE DIFFERENTLY!

> > `in` should accept only single level array
> 
> Done

really? It seems we do not validate/reject `in non-array`

> But we will keep equivalence of `=` and `in`. `in` also works with non-array value same as `=`

now ok, but `= arr` and `in non-arr` must throw, of course also for negative ops

Reply:
> we convert = null to is null, so = [null] MUST NEVER BEHAVE DIFFERENTLY!

This is handled by DSQL so if there is a problem it should be addressed there.
If any of the array values is null then convert it to `OR field IS NULL` instead of including it in the list

Reply:
> now ok, but `= arr` and `in non-arr` must throw, of course also for negative ops

but we should address this in Condition constructor to make the validation complete, right?

Reply:
> = arr and in non-arr must throw

It handles it equivalently now and it is okay. Basically now `=` === `in`

Reply:
that is bad, we should probably drop support for `=` with array and drop support for `in` with non-single-level-array